inc/user.class.php

```diff
@@ -3657,7 +3657,7 @@ static function getOrImportByEmail($email='') {
       $query = "SELECT `users_id` as id
                 FROM `glpi_useremails`
                 LEFT JOIN `glpi_users` ON (`glpi_users`.`id` = `glpi_useremails`.`users_id`)
-                WHERE `glpi_useremails`.`email` = '".stripslashes($email)."'
+                WHERE `glpi_useremails`.`email` = '".$DB->escape(stripslashes($email))."'
```

Comment:
forget my previous comment, I was thinking you add replace addslashes by escape...